### PR TITLE
add CONTRIBUTING.rst

### PR DESCRIPTION
**Pull request by TSC21**

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+To run the full suite of `tox`_ tests locally you'll need docker. Once you have docker installed
+and running do::
+
+    git submodule update --init --recursive
+    docker pull uavcan/toxic:py35-py38-sq
+    docker run --rm -it -v $PWD:/repo uavcan/toxic:py35-py38-sq
+    tox
+
+To run the c++ verification build you'll need to use a different docker container:
```
Comment (thirtytwobits) [inline CONTRIBUTING.rst:71]: omit c++ section

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+    docker pull uavcan/libuavcan:1.0
+    docker run --rm -it -v $PWD:/repo uavcan/libuavcan:1.0
+    ./.buildkite/verify_cpp.sh
+
+Files Generated by the Tests
```
Comment (thirtytwobits) [inline CONTRIBUTING.rst:77]: omit this section

File: CONTRIBUTING.rst (hunk 1 above)
Comment (TSC21) [reply CONTRIBUTING.rst]: Done

File: CONTRIBUTING.rst (hunk 2 above)
Comment (TSC21) [reply CONTRIBUTING.rst]: Done

**Pull request merged**